### PR TITLE
adds a label for initContainers, test and user-guide

### PR DESCRIPTION
**Pull request by sosan**
#### What type of PR is this?

/kind feature 
#### What this PR does / why we need it:
Add label for initContainers
for example: 
kompose.init-contaners.name: "init-service"
kompose.init-contaners.image: "busybox:1.28"
kompose.init-contaners.command: "['sh', '-c', 'until nslookup myservice.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for myservice; sleep 2; done']"

#### Which issue(s) this PR fixes:

Feature #1772

### CODE REVIEW

File: docs/user-guide.md [hunk 1]
```diff
@@ -467,6 +470,49 @@ services:
...
       kompose.volume.sub-path: pg-data
 ```
+
+
+- `kompose.init-contaners.name`  is used to specify the name of the Init Containers for a Pod [Init Container Name](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+
+For example:
+
+```yaml
+version: '2'
```
Comment (TessaIO) [inline docs/user-guide.md:480]: Let's refer to new versions
```suggestion
version: '3'
```

File: docs/user-guide.md [hunk 2]
```diff
@@ -467,6 +470,49 @@ services:
     labels:
       kompose.volume.sub-path: pg-data
 ```
+
+
```
Comment (TessaIO) [inline docs/user-guide.md:474]: small nit here:
```suggestion
```

File: docs/user-guide.md [hunk 3]
```diff
@@ -467,6 +470,49 @@ services:
...
+    labels:
+      kompose.init-contaners.name: "initcontainername"
+```
+
+- `kompose.init-contaners.image` defines image to use for the Init Containers [Init Container Image](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+
+For example:
+
+```yaml
+version: '2'
```
Comment (TessaIO) [inline docs/user-guide.md:493]: same as above

File: docs/user-guide.md [hunk 4]
```diff
@@ -467,6 +470,49 @@ services:
...
+      kompose.init-contaners.image: perl
+```
+
+
+- `kompose.init-contaners.command` defines the command that the Init Containers will run after they are started [Init Container Command](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+
+For example:
+
+```yaml
+version: '2'
```
Comment (TessaIO) [inline docs/user-guide.md:507]: same as above

File: pkg/transformer/kubernetes/k8sutils.go [hunk 5]
```diff
@@ -985,3 +985,49 @@ func reformatSecretConfigUnderscoreWithDash(secretConfig types.ServiceSecretConf
 
 	return newSecretConfig
 }
+
+//	looks for an initContainer resources and its passed as labels
+//
+// if there is no image, it does not fill the initContainer
+// https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
```
Comment (TessaIO) [inline pkg/transformer/kubernetes/k8sutils.go:992]: Let's conform with the Golang comments style here. I suggest:
```suggestion
// fillInitContainers looks for an initContainer resources and its passed as labels
// if there is no image, it does not fill the initContainer
// https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
```

File: pkg/transformer/kubernetes/k8sutils.go [hunk 6]
```diff
@@ -985,3 +985,49 @@ func reformatSecretConfigUnderscoreWithDash(secretConfig types.ServiceSecretConf
...
+	})
+}
+
+// parses a string containing comma-separated commands
+// returns a slice of strings or a single command and
+// example:
+// [ "bundle", "exec", "thin", "-p", "3000" ]
+//
+// example:
+// bundle exec thin -p 3000
```
Comment (TessaIO) [inline pkg/transformer/kubernetes/k8sutils.go:1016]: Same as above:
```suggestion
// parseContainerCommandsFromStr parses a string containing comma-separated commands
// returns a slice of strings or a single command and
// example:
// [ "bundle", "exec", "thin", "-p", "3000" ]
//
// example:
// bundle exec thin -p 3000
```

**Review by TessaIO — CHANGES_REQUESTED** [review]
@sosan Thanks for the PR. I put some comments below, and I'll do another round of review once my comments are addressed. One thing to add, can you add some e2e tests for this?

**Comment by sosan** [comment]
> @sosan Thanks for the PR. I put some comments below, and I'll do another round of review once my comments are addressed. One thing to add, can you add some e2e tests for this?

suggestions fixed
added test e2e for initcontainer

```
===> Starting test <===
convert::expect_success_and_warning: Running: 'kompose -f /home/runner/work/kompose/kompose/script/test/fixtures/initcontainer/compose.yaml convert --stdout --with-kompose-annotation=false' expected_output: '/home/runner/work/kompose/kompose/script/test/fixtures/initcontainer/output-k8s.yaml' expected_warning: ''
PASS: converted output matches
PASS: warning found: ''
```

**Review by cdrage — CHANGES_REQUESTED** [review]
Sorry for the delay!

Some feedback:
- contaners is spelt wrong
- To match all the other labels, it should be `kompose.init.container.name` and `kompose.init.container.image` and `kompose.init.container.command`.

Good addition! Again, apologizes for the delay.

File: pkg/loader/compose/utils.go [hunk 7]
```diff
@@ -87,6 +87,12 @@ const (
 	LabelCronJobConcurrencyPolicy = "kompose.cronjob.concurrency_policy"
 	// LabelCronJobBackoffLimit defines the job backoff limit
 	LabelCronJobBackoffLimit = "kompose.cronjob.backoff_limit"
+	// LabelInitContainerName defines name resource
+	LabelInitContainerName = "kompose.init-contaners.name"
+	// LabelInitContainerImage defines image to pull
+	LabelInitContainerImage = "kompose.init-contaners.image"
+	// LabelInitContainerComand defines comands
+	LabelInitContainerComand = "kompose.init-contaners.command"
```
Comment (cdrage) [inline pkg/loader/compose/utils.go:95]: LabelInitContainerComand should be spelled LabelInitContainerCommand

**Review by cdrage — CHANGES_REQUESTED** [review]

**Comment by sosan** [comment]
> Sorry for the delay!
> 
> Some feedback:
> 
> * contaners is spelt wrong
> * To match all the other labels, it should be `kompose.init.container.name` and `kompose.init.container.image` and `kompose.init.container.command`.
> 
> Good addition! Again, apologizes for the delay.

changed labels to kompose.init.container.name, kompose.init.container.image, kompose.init.container.command
fixed documentation, tests and e2e
also merged main with new commits to this branch

**Review by cdrage — APPROVED** [review]

**Pull request merged**